web/src/pages/api/public/media/upload-url.ts

```diff
@@ -0,0 +1,227 @@
+import { randomUUID } from "crypto";
+
+import { env } from "@/src/env.mjs";
+import { getFileExtensionFromContentType } from "@/src/features/media/server/getFileExtensionFromContentType";
+import { getMediaStorageServiceClient } from "@/src/features/media/server/getMediaStorageClient";
+import {
+  GetMediaUploadUrlQuerySchema,
+  GetMediaUploadUrlResponseSchema,
+  MediaContentType,
+} from "@/src/features/media/validation";
+import { createAuthedAPIRoute } from "@/src/features/public-api/server/createAuthedAPIRoute";
+import { withMiddlewares } from "@/src/features/public-api/server/withMiddlewares";
+import {
+  ForbiddenError,
+  InternalServerError,
+  InvalidRequestError,
+} from "@langfuse/shared";
+import { prisma } from "@langfuse/shared/src/db";
+import { instrumentAsync } from "@langfuse/shared/src/server";
+
+export default withMiddlewares({
+  POST: createAuthedAPIRoute({
+    name: "Get Media Upload URL",
+    bodySchema: GetMediaUploadUrlQuerySchema,
+    responseSchema: GetMediaUploadUrlResponseSchema,
+    successStatusCode: 201,
+    fn: async ({ body, auth }) =>
+      instrumentAsync({ name: "create-media-upload-url" }, async () => {
+        if (auth.scope.accessLevel !== "all") throw new ForbiddenError();
+
+        const { projectId } = auth.scope;
+        const {
+          contentType,
+          contentLength,
+          sha256Hash,
+          traceId,
+          observationId,
+          field,
+        } = body;
+
+        if (contentLength > env.LANGFUSE_S3_MEDIA_MAX_CONTENT_LENGTH)
+          throw new InvalidRequestError(
+            `File size must be less than ${env.LANGFUSE_S3_MEDIA_MAX_CONTENT_LENGTH} bytes`,
+          );
+
+        const { mediaId, uploadUrl } = await prisma.$transaction(async (tx) => {
```

Comment:
why do we need this upsert logic? Would it be enough to just insert? I am worried that we will maintain this table in CH in the future and will run into the same update issues we have right now